app/controllers/coronavirus_form/support_address_controller.rb

```diff
@@ -7,15 +7,25 @@ class CoronavirusForm::SupportAddressController < ApplicationController
     postcode
   ].freeze
 
+  def show
+    @form_responses = session.to_hash.with_indifferent_access
+    session[:support_address] = {}
+    respond_to do |format|
+      format.html { render controller_path }
+    end
+  end
+
   def submit
+    given_address = {
+      building_and_street_line_1: strip_tags(params[:building_and_street_line_1]&.strip).presence,
+      building_and_street_line_2: strip_tags(params[:building_and_street_line_2]&.strip).presence,
+      town_city: strip_tags(params[:town_city]&.strip).presence,
+      county: strip_tags(params[:county]&.strip).presence,
+      postcode: strip_tags(params[:postcode]&.gsub(/[[:space:]]+/, "")).presence,
+    }
+
     @form_responses = {
-      support_address: {
-        building_and_street_line_1: strip_tags(params[:building_and_street_line_1]&.strip).presence,
-        building_and_street_line_2: strip_tags(params[:building_and_street_line_2]&.strip).presence,
-        town_city: strip_tags(params[:town_city]&.strip).presence,
-        county: strip_tags(params[:county]&.strip).presence,
-        postcode: strip_tags(params[:postcode]&.gsub(/[[:space:]]+/, "")).presence,
-      },
+      support_address: helpers.compare_and_return(session[:support_address].symbolize_keys, given_address),
```

Comment:
I'm not really clear on the intent of the `compare_and_return` method.

Reply:
Fixed! See code.